File: app/src/main/java/com/mnj/java_solutions/linkedlist/DeleteAlternateNodes.java
package com.mnj.java_solutions.linkedlist;

public class DeleteAlternateNodes {

    public void deleteNodes() {
        ListNode node = new ListNode();
        ListNode.Node head = node.insert();

        ListNode.Node temp = head;

        while (temp != null && temp.next != null) {
            temp.next = temp.next.next;
            temp = temp.next;
        }

        System.out.println("==>> After operation.....");
        node.display(head);
    }
}

File: app/src/main/java/com/mnj/java_solutions/linkedlist/DeleteAlternateNodes.kt
package com.mnj.java_solutions.linkedlist

import com.mnj.java_solutions.linkedlist.ListNode

class DeleteAlternateNodes {
    fun deleteNodes() {
        val node = ListNode()
        val head = node.insert()
        var temp = head
        while (temp?.next != null) {
            temp.next = temp.next.next
            temp = temp.next
        }
        println("==>> After operation.....")
        node.display(head)
    }
}